Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.cs

```diff
@@ -96,6 +97,7 @@ public override void Initialize()
         SubscribeLocalEvent<GunComponent, ExaminedEvent>(OnExamine);
         SubscribeLocalEvent<GunComponent, CycleModeEvent>(OnCycleMode);
         SubscribeLocalEvent<GunComponent, HandSelectedEvent>(OnGunSelected);
+        SubscribeLocalEvent<GunComponent, ItemWieldedEvent>(OnGunWielded);
```

Comment:
Why isn't this event subscribed on `GunRequiresWieldComponent`? That's the actual component involved.

Reply:
Because `GunComponent.NextFire` is access-restricted to `SharedGunSystem`.

I can make a publicly available function to modify `NextFire` if that's the better way to go about it, though it would probably also involve refactoring `OnGunMelee` which does the same thing (but with `MeleeWeaponComponent`) for the same reason.

Reply:
Yes, do that.

Reply:
Turns out the thing `OnGunMelee` was doing was a bit different. Anyhow, changed it to its own public function now.